tests/test_ingestion.py

```diff
@@ -122,34 +127,43 @@ def testDataIngest(self):
     def testCalibIngest(self):
         """Test that ingesting calibrations adds them to a repository.
         """
-        calibData = [{'type': 'bias', 'file': 'bias.fits.gz', 'filter': 'None'},
-                     {'type': 'flat', 'file': 'flat_fg.fits.gz', 'filter': 'g'},
-                     {'type': 'flat', 'file': 'flat_fr.fits.gz', 'filter': 'r'},
-                     ]
         files = [os.path.join(IngestionTestSuite.testData, datum['type'], datum['file'])
-                 for datum in calibData]
+                 for datum in IngestionTestSuite.calibData]
 
         self._task._doIngestCalibs(self._repo, self._calibRepo, files)
 
         butler = self._calibButler()
-        for datum in calibData:
+        for datum in IngestionTestSuite.calibData:
             self.assertTrue(butler.datasetExists(datum['type'], filter=datum['filter']))
             # queryMetadata does not work on calibs
         self.assertFalse(butler.datasetExists('flat', filter='z'))
 
-    @unittest.skip("Ingestion functions cannot handle empty file lists, see DM-13835")
     def testNoFileIngest(self):
-        """Test that attempts to ingest nothing do nothing.
+        """Test that attempts to ingest nothing raise an exception.
         """
         files = []
 
-        self._task._doIngest(self._repo, files, [])
-        self._task._doIngestCalibs(self._repo, self._calibRepo, files)
+        with self.assertRaises(RuntimeError):
+            self._task._doIngest(self._repo, files, [])
+        with self.assertRaises(RuntimeError):
+            self._task._doIngestCalibs(self._repo, self._calibRepo, files)
 
         butler = self._calibButler()
         self.assertTrue(_isEmpty(butler, 'raw'))
 
-    # TODO: add unit test for _doIngest(..., badFiles) once DM-13835 resolved
+    def testBadFileIngest(self):
+        """Test that ingestion of raw data ignores blacklisted files.
+        """
+        badFiles = ['raw_v2_fg.fits.gz']
+
+        testDir = os.path.join(IngestionTestSuite.testData, 'raw')
+        files = [os.path.join(testDir, datum['file']) for datum in IngestionTestSuite.rawData]
+        self._task._doIngest(self._repo, files, badFiles)
+
+        butler = self._rawButler()
+        for datum in IngestionTestSuite.rawData:
+            dataId = {'visit': datum['visit']}
+            self.assertEqual(butler.datasetExists('raw', dataId), datum['file'] not in badFiles)
```

Comment:
Ooog. I'm not too keen on this syntax (it was hard for me to parse), but I'm not sure of a better approach given the butler methods that exist.

Reply:
I could do the approach of `if in badFiles: self.assertFalse(...) else: self.assertTrue(...)`, but I'd find that ickier.